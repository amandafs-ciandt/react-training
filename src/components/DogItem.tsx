import React from 'react';
import _ from 'lodash';

import { List, Avatar, Badge } from 'antd';

const DogItem = ({ dog }) => {
  return (
    <List.Item style={{ padding: '0.5rem 1rem' }}>
      <List.Item.Meta
        avatar={<Avatar src={dog.image} />}
        title={_.upperFirst(dog.breed)}
      />
      <Badge count={dog.scolded} showZero />
    </List.Item>
  );
};

export default DogItem;
