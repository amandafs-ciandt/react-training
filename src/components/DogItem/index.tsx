import React from 'react';
import _ from 'lodash';

import {
  ListItem,
  ListItemAvatar,
  Avatar,
  ListItemText,
  ListItemSecondaryAction,
  Chip,
} from '@material-ui/core';

import { Dog } from '../../shared/types';

interface IProps {
  dog: Dog;
}

const DogItem = ({ dog }: IProps) => {
  return (
    <ListItem alignItems='center' button>
      <ListItemAvatar>
        <Avatar alt={dog.breed} src={dog.image} data-testid='dog-item-avatar' />
      </ListItemAvatar>
      <ListItemText
        primary={_.upperFirst(dog.breed)}
        data-testid='dog-item-breed'
      />
      <ListItemSecondaryAction>
        <Chip label={dog.scolded} data-testid='dog-item-scolded-value' />
      </ListItemSecondaryAction>
    </ListItem>
  );
};

export default DogItem;
