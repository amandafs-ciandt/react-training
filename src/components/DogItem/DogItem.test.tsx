import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import DogItem from '.';

const dog = {
  breed: 'bulldog',
  image: 'https://images.dog.ceo/breeds/bulldog-french/n02108915_4474.jpg',
  scolded: 3,
};

const setUpShallowRendering = (): ShallowWrapper => {
  return shallow(<DogItem dog={dog} />);
};

describe('DogItem', () => {
  let dogItemComponent: ShallowWrapper;

  beforeEach(() => {
    dogItemComponent = setUpShallowRendering();
  });

  it('should render correctly', () => {
    expect(dogItemComponent).toMatchSnapshot();
  });

  it('should display the dog information correctly', () => {
    const itemAvatar = dogItemComponent.find('[data-testid="dog-item-avatar"]');
    const itemTitle = dogItemComponent.find('[data-testid="dog-item-breed"]');
    const itemScoldedValue = dogItemComponent.find(
      '[data-testid="dog-item-scolded-value"]'
    );

    expect(itemAvatar.prop('src')).toEqual(dog.image);
    expect(itemTitle.prop('primary')).toEqual('Bulldog');
    expect(itemScoldedValue.prop('label')).toEqual(dog.scolded);
  });
});
