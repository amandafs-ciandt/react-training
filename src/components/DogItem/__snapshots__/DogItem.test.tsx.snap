// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogItem should render correctly 1`] = `ShallowWrapper {}`;
